src/shared/components/query/QueryContainer.tsx

```diff
@@ -1,152 +1,277 @@
+                                        (<DataTypePrioritySelector/>) :
+                                        (<MolecularProfileSelector/>)
+                                    }
+
+                                    {(this.store.selectableSelectedStudyIds.length > 0) && (
+                                        <CaseSetSelector/>
+                                    )}
+
+                                    <GeneSetSelector/>
+
+                                    {!!(this.store.isGenesetProfileSelected) && (
+                                        <GenesetsSelector/>
+                                    )}
+
+                                    {!!(this.store.forDownloadTab) && (
+                                        <span className={styles.downloadSubmitExplanation}>
+									        Clicking submit will generate a tab-delimited file containing your requested data.
+								        </span>
+                                    )}
+
+                                    {!!(this.store.forDownloadTab) && (
+                                        <LabeledCheckbox
+                                            labelProps={{className: styles.transposeDataMatrix}}
+                                            checked={this.store.transposeDataMatrix}
+                                            onChange={event => this.store.transposeDataMatrix = event.currentTarget.checked}
+                                        >
+                                            Transpose data matrix
+                                        </LabeledCheckbox>
+                                    )}
+                                    <FlexRow overflow={true} padded
+                                             className={classNames(styles.submitRow, 'posRelative')}>
+                                        <button style={{paddingLeft: 50, paddingRight: 50, marginRight: 50}}
+                                                disabled={!this.store.submitEnabled} className="btn btn-primary btn-lg"
+                                                onClick={() => this.handleSubmit()} data-test='queryButton'>
+                                            {!this.store.forDownloadTab ? "Submit Query" : "Download"}
+                                        </button>
+                                        <FlexCol>
+                                            {!!(this.store.submitError) && (
+                                                <span className={styles.errorMessage} data-test="oqlErrorMessage">
+										            {this.store.submitError}
+									            </span>
+                                            )}
+
+                                            {this.store.oqlMessages.map(msg => {
+                                                return (
+                                                    <span className={styles.oqlMessage}>
+												        <i className='fa fa-info-circle' style={{marginRight: 5}}/>
```

Comment:
some formatting issues here? wouldnt necessarily mention it besides that you're doing a reformat